src/plugins/data/common/search/poll_search.ts

```diff
@@ -35,14 +36,35 @@ export const pollSearch = <Response extends IKibanaSearchResponse>(
       })
     );
 
+    const reportInfo = {
+      strategy: strategy || 'default',
+      status: '',
+      timeTookMs: new Date().getTime(),
+      pollCount: 0,
+      resHitCount: 0,
+      resAggCount: 0,
+    };
+
     return from(search()).pipe(
       expand(() => timer(pollInterval).pipe(switchMap(search))),
       tap((response) => {
         if (isErrorResponse(response)) {
           throw response ? new Error('Received partial response') : new AbortError();
         }
       }),
+      tap(() => {
+        reportInfo.pollCount++;
+      }),
       takeWhile<Response>(isPartialResponse, true),
+      tap((response) => {
+        // need to make sure this only works on FE
+        reportInfo.timeTookMs = new Date().getTime() - reportInfo.timeTookMs;
+        reportInfo.status = isErrorResponse(response) ? 'err' : 'ok';
+        const { hits, aggregations } = response.rawResponse;
+        reportInfo.resHitCount = hits?.hits?.length ?? 0;
+        reportInfo.resAggCount = aggregations ? Object.keys(aggregations).length : 0;
+        customEvents?.reportCustomEvent('search-done', reportInfo);
```

Comment:
I don't think these properties are going to work since it does not meet FullStory's event property naming requirements (the type needs to be included in the keys): https://help.fullstory.com/hc/en-us/articles/360020623234#Custom%20Event%20Name%20Requirements

Is this being converted elsewhere that I missed?

Reply:
Thanks, I missed that in my quick look.